GitHub app and action release notes (#9543)
pantheon-systems/documentation: source/content/github-application.md

categories: [automate, workflows]
cms: [drupal, wordpress]
audience: [development]
product: [terminus]
integration: [--]
reviewed: "2025-04-15"

---

Pantheon's GitHub application directly integrates a GitHub repository with a Pantheon site.
This allows you to use GitHub as your code repository while still using Pantheon to run your site.

The GitHub application is currently in private Beta. [Request access for your Pantheon workspace here](https://docs.google.com/forms/d/e/1FAIpQLSf0vYrRbPQBxR-hT8kGJ4bEdYPtpkTtfDvPM89xD2dNZeqLqA/viewform).

Once turned, on this application accommodates a pull request workflow where a Multidev environment is created for each pull request. This allows you to test the code in the pull request before merging it.

![Deploying a PR to a Pantheon Multidev](../images/github-app/diagram--deploying-pr.png)

Merging pull requests, or pushing code to the main branch of the GitHub repository, will automatically deploy the code to the Pantheon development environment.

![Deploying main to Pantheon](../images/github-app/diagram--deploying-main.png)

## Creating a new site

The GitHub application works with newly created Pantheon sites and GitHub repositories.

> Before the application is made generally available, it will not work with existing Pantheon sites or GitHub repositories.

Author: ```suggestion
We will soon add the capacity to connect the application to pre-existing repositories and sites.
```

## Limitations prior to General Availability

The GitHub application is currently in private Beta.

We will address these limitaitons before the application is made generally available.

### Limited to new projects created by the Terminus

Activating the GitHub Application presently runs through Terminus that creates a new site on Pantheon and a new GitHub repository. The GitHub application does not yet support retrofitting existing GitHub repositories or Pantheon sites yet.
Follow [this issue](https://github.com/pantheon-systems/terminus/issues/2683) in the queue to find out when this limitation is removed.

### No On Server Development  (SFTP Mode)

New sites made with the GitHub Application do not support "[SFTP Mode](/guides/sftp)" which allows version controlled files to be altered via SFTP or simply by the CMS changing files, as is common with operations like "drush config-export." We know this limitation will stop some teams from using this application and [we are seeking feedback on how important it is to support this style of working when using 3rd party repositories](https://roadmap.pantheon.io/c/115-github-gitlab-and-bitbucket-integration).

> 

Author: ```suggestion

### Not yet compatible with GitHub Enterprise

The Pantheon GitHub Application is currently only compatible with repositories on GitHub.com, not repositories on GitHub Enterprise instances.

```

@DuncanSchouten @kporras07 Is this accurate ^

Reviewer: Kevin did some testing and the GitHub Application _does_ work with GitHub Enterprise as-is. For now, let's not advertise this.

Author: I'm confused. Should we have this disclaimer because we haven't tested thoroughly with Enterprise or do we not need this disclaimer because the testing showed it worked?

### No On Server Development  (SFTP Mode)

New sites made with the GitHub Application do not support "[SFTP Mode](/guides/sftp)" which allows version controlled files to be altered via SFTP or simply by the CMS changing files, as is common with operations like `drush config-export`. We know this limitation will stop some teams from using this application and [we are seeking feedback on how important it is to support this style of working when using 3rd party repositories](https://roadmap.pantheon.io/c/115-github-gitlab-and-bitbucket-integration).

## Support

> Before this functionality is made generally available, please direct questions and feedback to [the issue queue on the Terminus repository](https://github.com/pantheon-systems/terminus) rather than general support channels.

Author: ```suggestion
Before this functionality is made generally available, please direct questions and feedback to [the issue queue on the related Terminus plugin](https://github.com/pantheon-systems/terminus-repository-plugin) rather than general support channels.
```
